mgmt: mcumgr: Make img_mgmt structures packed

nordicjm opened the pull request:
Resolves an issue with processors that do not support unaligned memory
access when using img_mgmt functions, e.g. ARM Cortex M0, by marking
structures as packed.

Fixes #49066

de-nordic commented on subsys/mgmt/mcumgr/Kconfig:
```diff
@@ -15,6 +15,24 @@ module = MCUMGR
+config MCUMGR_ALIGNED_MEMORY_REQUIREMENT
+	bool
+	depends on MCUMGR_ALIGN_MEMORY
+	default y
```
Shouldn't this be done by `select` in `MCUMGR_ALIGN_MEMEORY`?

nordicjm replied:
Fixed

carlescufi reviewed (changes requested):
Please mark the struct as `packed` instead

nordicjm committed:
mgmt: mcumgr: Make img_mgmt structures packed

Resolves an issue with processors that do not support unaligned memory
access when using img_mgmt functions, e.g. ARM Cortex M0, by marking
structures as packed.

Signed-off-by: Jamie McCrae <jamie.mccrae@nordicsemi.no>
